src/NServiceBus.Core.Tests/DelayedDelivery/TimeoutManager/ExpiredTimeoutsPollerTests.cs

```diff
@@ -64,6 +67,30 @@ public void TearDown()
             Assert.AreEqual(currentTime + InMemoryTimeoutPersister.EmptyResultsNextTimeToRunQuerySpan, poller.NextRetrieval);
         }
 
+        [Test]
+        public async Task Polls_with_same_start_slice_when_dispatch_fails()
+        {
+            var failingDispatcher = new FailableDispatcher();
+            poller = new ExpiredTimeoutsPoller(timeouts, failingDispatcher, "test", breaker, () => currentTime);
+
+            RegisterNewTimeout(currentTime.Subtract(TimeSpan.FromMinutes(5)));
+
+            // fail first dispatch
+            failingDispatcher.DispatcherAction = _ => { throw new Exception("transport error"); };
+            Assert.ThrowsAsync<Exception>(() => poller.SpinOnce(CancellationToken.None));
+
+            // succeed second dispatch
+            var unicastTransportOperations = new List<UnicastTransportOperation>();
+            failingDispatcher.DispatcherAction = m =>
```

Comment:
A bit on the fence about the two setups of the dispatcher action. Since the dispatcher is primarily written just for this test, you could also hard code it in the dispatcher itself. So, for example, the dispatcher could be called `FirstCallFailsSecondCallThrowsDispatcher` and expose the collected transport operation. The downside of it is that it would hide it. Or just one setup of the DispatchAction with an internal behavior switch per call. But that is just a preference thing and I don't want to hold it up just because of that. More of a discussion point